### PR TITLE
Feature/sl 1551/search and find listing request

### PR DESCRIPTION
**Pull request by kpakur**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sandl/snlapi/controllers/HearingController.java [hunk 1]
```diff
@@ -35,4 +40,16 @@ public ResponseEntity assignHearingPartToSession(
         @RequestBody String assignment) {
         return eventsCommunicationService.makePutCall("/hearing/{hearingId}", assignment, hearingId);
     }
+
+    @PostMapping(path = "", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    public String searchHearings(@RequestParam(value = "isListed", required = false) Optional<Boolean> isListed,
+                                  @RequestParam(value = "page", required = false) Optional<Integer> page,
```
Comment (filipczyks) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlapi/controllers/HearingController.java]: not aligned

File: src/main/java/uk/gov/hmcts/reform/sandl/snlapi/controllers/HearingController.java [hunk 2]
```diff
@@ -35,4 +40,16 @@ public ResponseEntity assignHearingPartToSession(
...
+                                  @RequestParam(value = "size", required = false) Optional<Integer> size,
+                                    @RequestBody String searchCriteria) {
```
Comment (filipczyks) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlapi/controllers/HearingController.java]: not aligned

File: src/main/java/uk/gov/hmcts/reform/sandl/snlapi/controllers/HearingController.java [hunk 3]
```diff
@@ -35,4 +40,16 @@ public ResponseEntity assignHearingPartToSession(
...
+        String url = "/hearing" +  (isListed.isPresent() ? "?isListed=" + isListed.get() : "");
+        url += (page.isPresent() && size.isPresent()) ? "?page=" + page.get() + "&size=" + size.get() : "";
```
Comment (filipczyks) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlapi/controllers/HearingController.java]: page is always concatenated with ?, I think you will lose isListed value if you add page.
Query string will look like ?isListed=?page=&size=

File: src/main/java/uk/gov/hmcts/reform/sandl/snlapi/controllers/HearingController.java (hunk 1 above)
Comment (kpakur) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlapi/controllers/HearingController.java]: better now?

File: src/main/java/uk/gov/hmcts/reform/sandl/snlapi/controllers/HearingController.java (hunk 2 above)
Comment (kpakur) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlapi/controllers/HearingController.java]: better now?

File: src/main/java/uk/gov/hmcts/reform/sandl/snlapi/controllers/HearingController.java (hunk 3 above)
Comment (kpakur) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlapi/controllers/HearingController.java]: done

**Pull request merged**